src/cli/cluster/base_path_proxy.js

```diff
@@ -18,123 +18,47 @@
  */
 
 import { Server } from 'hapi';
-import { notFound } from 'boom';
-import { map, sample } from 'lodash';
-import { map as promiseMap, fromNode } from 'bluebird';
-import { Agent as HttpsAgent } from 'https';
-import { readFileSync } from 'fs';
-
-import { setupConnection } from '../../server/http/setup_connection';
-import { registerHapiPlugins } from '../../server/http/register_hapi_plugins';
+import { createBasePathProxy } from '../../core';
 import { setupLogging } from '../../server/logging';
 
-const alphabet = 'abcdefghijklmnopqrztuvwxyz'.split('');
+export async function configureBasePathProxy(config) {
```

Comment:
So, this is the function that takes in a config, creates a basePathProxy, and then runs .configure on it using the input config. It retuns a configured basePathProxy.

What do you think of naming this configure_base_path_proxy or configured_base_path_proxy? I think base_path_proxy as a name should probably refer to the server, which is in src/core/server/http/base_path_proxy_server.js (which we appended `server` to, but could work even without that).

Also, what about a simple test file in this directory for this function?

Reply:
Yeah, +1 to both your ideas, will handle them.